hibernate-core/src/main/java/org/hibernate/proxy/AbstractLazyInitializer.java

```diff
@@ -273,7 +273,11 @@ protected void prepareForPossibleLoadingOutsideTransaction() {
 		if ( session != null ) {
 			allowLoadOutsideTransaction = session.getFactory().getSessionFactoryOptions().isInitializeLazyStateOutsideTransactionsEnabled();
 
-			if ( allowLoadOutsideTransaction && sessionFactoryUuid == null ) {
+			if ( sessionFactoryUuid == null ) {
+				//we're going to need the UUID even if we the SessionFactory configuration doesn't
+				// allow any operations on it, as we need it to match deserialized objects with
+				// the originating SessionFactory: at very least it's useful to actually get
+				// such configuration, so to know if such operation isn't allowed or configured otherwise.
```

Comment:
Looks like you've changed the purpose of this method; maybe it should be renamed?

I don't know if this class is API or "looks like API but actually is not", though...

Reply:
Good catch, I've amended the javadoc to clarify.